packages/react-model-ui/src/views/common/EntityAttributesDataGrid.tsx

```diff
@@ -0,0 +1,290 @@
+/********************************************************************************
+ * Copyright (c) 2023 CrossBreeze.
+ ********************************************************************************/
+import { EntityAttribute } from '@crossbreeze/protocol';
+import { useModel, useModelDispatch } from '../../ModelContext';
+import * as React from 'react';
+import {
+    DataGrid,
+    GridActionsCellItem,
+    GridCellEditStartParams,
+    GridCellModes,
+    GridCellParams,
+    GridColDef,
+    GridEditCellProps,
+    GridEditInputCell,
+    GridPreProcessEditCellProps,
+    GridRenderEditCellParams,
+    GridRowId,
+    GridRowModel,
+    GridRowsProp,
+    GridToolbarContainer,
+    MuiEvent,
+    useGridApiRef
+} from '@mui/x-data-grid';
+import { FormControl, MenuItem, Select, SelectChangeEvent, Tooltip, TooltipProps, tooltipClasses, styled, Box } from '@mui/material';
+import Button from '@mui/material/Button';
+import AddIcon from '@mui/icons-material/Add';
+import DeleteIcon from '@mui/icons-material/Delete';
+import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';
+import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
+
+export function EntityAttributesDataGrid(): React.ReactElement {
+    // Context variables to handle model state.
+    const model = useModel();
+    const gridApiRef = useGridApiRef();
+    const dispatch = useModelDispatch();
+    const [currentEdit, setCurrentEdit] = React.useState({} as CurrentEdit);
+
+    // Callback for when the user stops editing a cell.
+    const handleRowUpdate = (updatedRow: GridRowModel, originalRow: GridRowModel): GridRowModel => {
+        // Handle change of name property.
+        dispatch({
```

Comment:
In the previous property view attributes grid, there was a validation that checks if a name is empty and if it is, it sets that row as erroneous by applying a different CSS class (`entity-attribute-error-row`). Has this been removed on purpose? I think it is fine since we really didn't do any particular client-side validation so we should properly think about it but I just wanted to point that out.

Reply:
I indeed purposely remoded it, because it was some custom code which required new state to be maneged. While when I was reading the mui-datagrid docs I saw there is a preprocessing hook for every field.

So I implemented one for the name here: https://github.com/CrossBreezeNL/crossmodel/pull/32/commits/88440351fd9b915bb4e5a9b3ea13e2c3eec4b428#diff-db8e8a38db340fd118288147c3126f021b6261c77da14bfa54d34fa540cfd304R135.

It now shows a tooltip on the field with the error, the updatedRow event is fired.